planter/planter.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+)
+
+// TODO(bentheelder): improve usage documentation
+// TODO(bentheelder): how can we support DOCKER_EXTRA in a sane fashion?
```

Comment:
This is tricky since DOCKER_EXTRA has just been an extra substring of the shell command for the docker run.
The best alternative I have is to not make planter.go a drop-in replacement and instead support:

`TAG=0.7.0-1 DRY_RUN= go run planter/planter.go bazel test //... -- -some-docker-arg -some-other-docker-arg`

This would be a little backwards perhaps, but it would allow the common use case to be:
`go run planter/planter.go bazel test //...`

Reply:
... except that you need to supply bazel with `--`.
Another somewhat gross option I had was to parse DOCKER_EXTRA="--foo 'bar --bar'" into an arg array preserving quoting, or to basically just implement `system("<shell command here>")`.

Perhaps it could be `$(DOCKER_EXTRA=-some-arg go run planter/planter.go)`, with planter.go just printing the command to execute (IE always "dry mode").